File: Src/Sst/LuaTemplate/Lib/Controller.lua
﻿local classes = require "Classes"

local Controller = classes.class()

function Controller:init(child)
	self.IController = LuaProxy:CreateController(child)
end

function Controller:Register(key,func)
	local id = self.IController:Register(key,func)

	if self.MethodIds == nil then
		self.MethodIds = id
	else
		self.MethodIds = self.MethodIds .. ";" .. id
	end
end

function Controller:Invoke(methodId,peer,callback,...)
	self.IController:Invoke(methodId,peer,callback,...)
end

return Controller

File: Src/Sst/PythonTemplate/Lib/Controller.py
import clr
import sys

clr.AddReference("Sss")

from Sss.SssScripts.Python import PythonProxy

class Controller:
	def __init__(self):
		self.MethodIds = ""
		self.IController = PythonProxy.CreateControler(self)

	def Register(self, key,value):
		id = self.IController.Register(key,value,sys.PythonHelper)
		if self.MethodIds == "":
			self.MethodIds = id
		else:
			self.MethodIds = self.MethodIds + ";" + id

	def Invoke(self,id,peer,callback,*args):
		self.IController.Invoke(id,peer,callback,args)